src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -839,7 +840,9 @@ namespace
             return StatusCode::InvalidConfigFile;
         }
 
-        // [TODO] Validate the current context is acceptable for this request (frameworks)
+        // Validate the current context is acceptable for this request (frameworks)
+        if (!fx_resolver_t::is_config_compatible_with_frameworks(app_config, existing_context->fx_versions_by_name))
+            return StatusCode::CoreHostIncompatibleConfig;
```

Comment:
Do we always want a hard fail here?

Reply:
At least for now I think yes. I could imagine that in the future we allow the native app to override this decision, but for now I think it's safer to not go there yet.